### PR TITLE
Ape 0.8

### LINKED ISSUE #315: Upgrade ape dependencies to > 0.8
We are stuck in the past with our `ape` dependencies. Currently using 0.7.13 for `eth-ape` but the latest is 0.8.12.

### PR DESCRIPTION
**Pull request by derekpierre**
**Type of PR:**
- [ ] Bugfix
- [ ] Feature
- [ ] Documentation
- [x] Other

**Required reviews:** 
- [ ] 1
- [x] 2
- [ ] 3

**What this does:**
> High-level idea of the changes introduced in this PR. 
> List relevant API changes (if any), as well as related PRs and issues.

- Relocks dependencies
- Upgrade to latest version of `ape` 0.8.x and updates relevant ape plugins

Closes #315 

**Issues fixed/closed:**
> - Fixes #...

**Why it's needed:**
> Explain how this PR fits in the greater context of the NuCypher Network.
> E.g., if this PR address a `nucypher/productdev` issue, let reviewers know!

**Notes for reviewers:**
> What should reviewers focus on? 
> Is there a particular commit/function/section of your PR that requires more attention from reviewers?

### CODE REVIEW

**Review by manumonti — APPROVED** [review]
Thanks for taking care of this! 🙌

File: ape-config.yaml [hunk 1]
```diff
@@ -27,17 +27,10 @@ dependencies:
 solidity:
   version: 0.8.23
   evm_version: paris
-  import_remapping:
```
Comment (cygnusv) [inline ape-config.yaml:30]: yay!

File: contracts/test/proxy/Destroyable.sol [hunk 2]
```diff
@@ -1,40 +0,0 @@
...
-pragma solidity ^0.8.0;
-
-
-import "../../contracts/proxy/Upgradeable.sol";
-
-
-/**
-* @dev Contract that could be destroyed by selfdestruct
-*/
-contract Destroyable is Upgradeable {
```
Comment (cygnusv) [inline contracts/test/proxy/Destroyable.sol:12]: I didn't remember this, at all

**Review by cygnusv — APPROVED** [review]

File: contracts/test/proxy/Destroyable.sol (hunk 2 above)
Comment (derekpierre) [reply contracts/test/proxy/Destroyable.sol:12]: I didn't know it existed 😅 . 

A test failed after the update to `ape`, and I checked with @vzotova, who indicated that it was never used.

**Pull request merged**